forem/forem
# Support for new ad placement locations on home page feed

## What type of PR is this? (check all applicable)


## Description

Tasks finished from issues:

- [x] feed_first - Before all home page posts
- [x] feed_second - Between 2nd and 3rd posts in the feed
- [x] feed_third - Between 7th and 8th posts in the feed
- [x] these options should be available at /admin/customization/display_ads/new
- [ ] these options should appear in API documentation for display_ads **( Implemented in #19306)**
- [x] remove hard-coded sign_in_invitation (aka CTA)
- [x] remove [event](https://github.com/forem/forem/blob/09291b420515d265eca031b431a5937bc63c3bb3/app/javascript/packs/homePage.jsx#L135) as we won't be tracking clicks on this CTA.

## Related Tickets & Documents



- Related Issue #19225

## QA Instructions, Screenshots, Recordings

[Figma](https://www.figma.com/file/9UfSjNYJs745w3Q8JCUqsL/Small-Wins---%E2%80%A8Core-Experience-%E2%9A%A1%EF%B8%8F?node-id=1623-7063&t=fvJIv0Rh0lEk374A-0)

Loom Recording: https://www.loom.com/share/92f295adf4c34a89a20efb30e4cdf878
There was one minor bug mentioned in above video it has been fixed: https://www.loom.com/share/a5a4cf28843e4215b86833b133ca9a93

**How to test?**

1. Checkout in this branch and sign in as admin locally
2. Create three new billboards for position "Home Feed First", "Home Feed Second" and "Home Feed Third"
3. And either sign-out or open home page in incognito mode
4. Check the three billboards that you created at various positions.

**Output**

<img width="1512" alt="Screenshot 2023-04-07 at 3 01 34 PM" src="https://user-images.githubusercontent.com/9396084/230588248-f183ce53-6700-4046-926e-bfd42f71b95a.png">
<img width="607" alt="Screenshot 2023-04-07 at 3 01 49 PM" src="https://user-images.githubusercontent.com/9396084/230588270-11d38892-fe5a-4e2b-b807-7c6e4853557a.png">
<img width="618" alt="Screenshot 2023-04-07 at 3 01 59 PM" src="https://user-images.githubusercontent.com/9396084/230588276-a78ba461-f5e9-4184-a62e-12a480cd8c2f.png">



### UI accessibility concerns?

No

## Added/updated tests?

- [x] No, need help with exactly where can I add these tests and also add display ads to the feed.

## [optional] Are there any post deployment tasks we need to perform?
